qiskit/transpiler/passmanager.py

```diff
@@ -33,49 +38,98 @@
 class PassManager(BasePassManager):
     """Manager for a set of Passes and their scheduling during transpilation."""
 
-    PASS_RUNNER = RunningPassManager
-
     def __init__(
         self,
-        passes: PassSequence | None = None,
+        passes: OptimizerTask | list[OptimizerTask] = None,
         max_iteration: int = 1000,
     ):
-        """Initialize an empty `PassManager` object (with no passes scheduled).
+        """Initialize an empty pass manager object.
 
         Args:
-            passes: A pass set (as defined in :py:func:`qiskit.transpiler.PassManager.append`)
-                to be added to the pass manager schedule.
+            passes: A pass set to be added to the pass manager schedule.
             max_iteration: The maximum number of iterations the schedule will be looped if the
                 condition is not met.
         """
-        super().__init__(passes, max_iteration)
-        self.property_set = None
+        super().__init__(max_iteration=max_iteration)
+
+        # Override type
+        self._flow_controller = RunningPassManager()
```

Comment:
If we have to override something in the base class like this, it feels like the interface and responsibilities of the different components might not be fully aligned correctly.  What's the purpose of having the base class instantiate a flow controller if we know that the only real use of the base class needs to replace it?

Reply:
This is just for backward compatibility. `RunningPassManager` is basically equivalent to `FlowControllerLinear` and in principle we don't need to override the baseclass flow controller like this. However, to avoid some API break, we still need running pass manager for circuit pass manager.

Reply:
Ah, that's ok.  Would you mind putting in a comment to say that it's just for legacy reasons?

Do we have a plan for what needs to happen before we could remove it again?

Reply:
Yes, done in [6211a5d](https://github.com/Qiskit/qiskit/pull/10127/commits/6211a5d2978af9091eb5f474273b5810828b6d68)